gateway/ln-gateway/src/lib.rs

```diff
@@ -257,6 +257,11 @@ pub struct Gateway {
     // incoming HTLCs.
     clients: FederationToClientMap,
 
+    /// Joining new Federation is protected by this lock to prevent trying to
```

Comment:
Would you prefer to first land/rebase https://github.com/fedimint/fedimint/pull/4295?

Reply:
Same question for https://github.com/fedimint/fedimint/pull/4294

Reply:
We should review and land in parallel. They are the exact same change so there's not conflict. And this change might take a while as I keep discovering things to fix.

Reply:
always be landing, whatever you can, whenever you can. :)